### ISSUE TITLE
socket.io disconnects arbitrarily

### ISSUE THREAD

**Issue by rafaelcr** [hirosystems/stacks-blockchain-api#1464]
From Discord:

We have been using the socket.io connection successfully for months to listen for events on our own mocknet instance. But for the last few days, after a single successful event, it disconnects with an io server disconnect reason. And it doesn't restart. 
Only thing that changed was updating clarinet to its latest version, but I'm not sure how that would affect it. 
The problem started while we were still running blockchain-api-client v5.0.0 but its still the same after an upgrade to v6.2.1.

```js
import { StacksApiSocketClient } from '@stacks/blockchain-api-client';
import { io as ioClient } from 'socket.io-client';

export default function startObserver() {
  const socket = ioClient(IOCLIENT_URI, {
    transports: [ "websocket" ]
  });
  const stacksSocket = new StacksApiSocketClient(socket);

  // Had to add this so it reconnects:
  stacksSocket.socket.on('disconnect', async (reason) => {
    console.log(`[Stacks] Disconnecting, reason: ${reason}`);
    stacksSocket.socket.connect();
  });

  // Subscribing to contract's txs
  stacksSocket.subscribeAddressTransactions(contractFullName);

  // Handling incoming txs
  stacksSocket.socket.on('address-transaction', async (address, txWithTransfers) => {
    const _tx = txWithTransfers.tx;
    const _successful = _tx.tx_status === 'success';
    if (!_successful) {
      console.log(`[Stacks] Failed tx...: ${_tx.tx_id}`);
      return;
    }

    // This is just a fetch to the Stacks API:
    const txInfo = await fetchTXInfo(_tx.tx_id);

    // This could be just a console.log:
    handleTx(txInfo);
  })
}
```

To recap, after every successful hearing of an 'address-transaction', it disconnects. The  reason is always just: io server disconnect. IOCLIENT_URI has no secure wss, just ws, but this hasn't been a problem in the past.

**Comment by zone117x** [comment]
This was fixed by https://github.com/hirosystems/stacks-blockchain-api/pull/1800 AFAIK
